eoscompanion/format.py

```diff
@@ -37,14 +37,14 @@ def format_app_icon_uri(icon_name, device_uuid):
     )
 
 
-def format_thumbnail_uri(application_id, model, device_uuid):
+def format_thumbnail_uri(application_id, thumbnail_uri, device_uuid):
     '''Format a uri to get an icon for an app.'''
     return format_uri_with_querystring(
         '/v1/content_data',
         deviceUUID=device_uuid,
         applicationId=application_id,
-        contentId=urllib.parse.urlparse(model['thumbnail_uri']).path[1:]
-    ) if model.get('thumbnail_uri', None) else None
+        contentId=urllib.parse.urlparse(thumbnail_uri).path[1:]
+    ) if thumbnail_uri else None
```

Comment:
If this change is necessary, can you split it out into a separate commit?

Reply:
For what its worth, you could cherry-pick https://github.com/endlessm/eos-companion-app-integration/pull/95/commits/ebdce0c6d15fbd888d5a78ee272624e610f586db which should do what you need here.